src/main/java/duke/DukeList.java

```diff
@@ -0,0 +1,258 @@
+package duke;
+
+
+import duke.exceptions.DukeException;
+import duke.exceptions.DukeInvalidCommandException;
+import duke.exceptions.DukeInvalidDescriptionException;
+import duke.exceptions.DukeNoDateException;
+import duke.exceptions.DukeNoDescriptionException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.TaskType;
+import duke.tasks.Todo;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * DukeList class for Duke.
+ * Stores tasks.
+ */
+public class DukeList {
+
+    private static final int CAPACITY = 100;
+    private final List<Task> taskList;
+    private final Storage store = new Storage("data", "duke.txt");
+
+
+    /**
+     * Constructor for DukeList.
+     */
+    public DukeList() {
+        this.taskList = new ArrayList<>(CAPACITY);
```

Comment:
As it is possible to initialise the taskList without a defined capacity, would it be better to just leave it undefined so that there is no limit to the number of tasks that will be added in?

Reply:
I actually wanted to limit the capacity to improve runtime, but thanks for the feedback!